### PR TITLE
entities owner deletion part 2

### PR DESCRIPTION
**Pull request by anandjakhaniya**
- added logic when deleting the entity
- added test cases for the type of condition

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java [hunk 1]
```diff
@@ -116,16 +116,39 @@ public DatasetVersionServiceImpl(
...
       accessibleDatasetVersionIds.addAll(requestedDatasetVersionIds);
     } else {
-      allowedDatasetIds =
+      allowedDatasetVersionIds =
           roleService.getSelfAllowedResources(
               ModelDBServiceResourceTypes.DATASET, modelDBServiceActions);
       // Validate if current user has access to the entity or not
-      allowedDatasetIds.retainAll(requestedDatasetVersionIds);
-      accessibleDatasetVersionIds.addAll(allowedDatasetIds);
+      allowedDatasetVersionIds.retainAll(datasetIdSet);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java]: datasetIdSet ? is this right? if so was there a bug to begin with?

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java]: it was a typo. fixed

Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java]: I do not see this changed

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java [hunk 2]
```diff
@@ -116,16 +116,39 @@ public DatasetVersionServiceImpl(
...
+      }
     }
     return accessibleDatasetVersionIds;
   }
 
+  public List<String> getAccessibleDatasetVersionIDsByAction(
+      List<String> requestedDatasetVersionIds, ModelDBServiceActions modelDBServiceActions) {
+    // Validate if current user has access to the entity or not
+    if (requestedDatasetVersionIds.size() == 1) {
+      roleService.isSelfAllowed(
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java:137]: let us say a user has delete privilages on Dataset and deos not have a role binding at dataset version . how will this call to isSelfAllowed  working?

**Review by ravishetye — CHANGES_REQUESTED** [review]
we can follow up on zoom but this requires followup

**Review by ravishetye — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java:137]: resolved conversation over zoom

**Pull request merged**